src/rluni/robot/RWIP.py

```diff
@@ -223,6 +228,37 @@ async def control_loop(self, shutdown_event):
             loop_delay = self.precise_delay_until(end_time)
             loop_period = time.time() - loop_start_time
 
+    def precise_delay_until(self, end_time):
+        """
+        Sleep until the specified end time, then return the delay time (overshoot)
+        """
+        # Lower accuracy sleep loop
+        while True:
+            now = time.time()
+            remaining = end_time - now
+            if remaining <= 0.0007:
+                break
+            time.sleep(remaining / 2)
+
+        # Busy-while loop to get accurate cutoff at end
+        while True:
+            delay = time.time() - end_time
+            if delay >= 0:
+                return delay
+
+    """ Shutdown Methods """
+
+    async def shutdown(self):
```

Comment:
For the future, shutdown should be changed to shutdown all motors, not for this PR. Just a note.

Reply:
Sure, the idea for this function is to clean up everything needed, should have the other motor added